keras_nlp/models/roberta/roberta_preprocessing.py

```diff
@@ -19,16 +19,158 @@
 from tensorflow import keras
 
 from keras_nlp.tokenizers.byte_pair_tokenizer import BytePairTokenizer
+from keras_nlp.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class RobertaTokenizer(BytePairTokenizer):
+    """A RoBERTa tokenizer using Byte-Pair Encoding subword segmentation.
+
+    This tokenizer class will tokenize raw strings into integer sequences and
+    is based on `keras_nlp.tokenizers.BytePairTokenizer`. Unlike the
+    underlying tokenizer, it will check for all special tokens needed by RoBERTa
+    models and provides a `from_preset()` method to automatically download
+    a matching vocabulary for a RoBERTa preset.
+
+    This tokenizer does not provide truncation or padding of inputs. It can be
+    combined with a `keras_nlp.models.RobertaPreprocessor` layer for input
+    packing.
+
+    If input is a batch of strings (rank > 0), the layer will output a
+    `tf.RaggedTensor` where the last dimension of the output is ragged.
+
+    If input is a scalar string (rank == 0), the layer will output a dense
+    `tf.Tensor` with static shape `[None]`.
+
+    Args:
+        vocabulary: string or dict, maps token to integer ids. If it is a
+            string, it should be the file path to a json file.
+        merges: string or list, contains the merge rule. If it is a string,
+            it should be the file path to merge rules. The merge rule file
+            should have one merge rule per line. Every merge rule contains
+            merge entities separated by a space. Please refer to this example:
+            https://storage.googleapis.com/keras-nlp/models/roberta_base/merges.txt.
+
+    Examples:
+
+    Batched inputs.
+    >>> vocab = {
+    ...     "<s>": 0,
+    ...     "<pad>": 1,
+    ...     "</s>": 2,
+    ...     "reful": 3,
+    ...     "gent": 4,
+    ...     "Ġafter": 5,
+    ...     "noon": 6,
+    ...     "Ġsun": 7,
+    ... }
+    >>> merges = ["Ġ a", "Ġ s", "r e", "f u", "g e", "n t"]
+    >>> merges += ["e r", "n o", "o n", "i g", "h t"]
+    >>> merges += ["Ġs u", "Ġa f", "ge nt", "no on", "re fu"]
+    >>> merges += ["Ġsu n", "Ġaf t", "refu l", "Ġaft er"]
+    >>> inputs = [" afternoon sun", "refulgent sun"]
+    >>> tokenizer = keras_nlp.models.RobertaTokenizer(
+    ...     vocabulary=vocab,
+    ...     merges=merges,
+    ... )
+    >>> tokenizer(inputs)
+    <tf.RaggedTensor [[5, 6, 7], [3, 4, 7]]>
+
+    Unbatched input.
+    >>> vocab = {
+    ...     "<s>": 0,
+    ...     "<pad>": 1,
+    ...     "</s>": 2,
+    ...     "reful": 3,
```

Comment:
maybe for these later two examples, just to keep the size down, re could remove "refulgent" and the associated merge rules?

If we could retool these so the vocab fits on one line that would be great. We want to focus to be on the usage, which get's a little lost visually here.